src/components/AssignProductDialog/ModalProductFilterProvider.tsx

```diff
@@ -0,0 +1,61 @@
+import { createContext, FC, ReactNode, useContext } from "react";
+
+import { ConditionalFilterContext } from "../ConditionalFilter/context/context";
+import { useModalProductFilter, UseModalProductFilterResult } from "./useModalProductFilter";
+
+type ModalProductFilterContextValue = UseModalProductFilterResult;
+
+const ModalProductFilterContext = createContext<ModalProductFilterContextValue | null>(null);
+
+export interface ProductTypeConstraint {
+  id: string;
+  name: string;
+}
+
+export interface InitialConstraints {
+  productTypes?: ProductTypeConstraint[]; // ProductType constraints from reference attribute
+}
+
+export interface ModalProductFilterProviderProps {
+  children: ReactNode;
+  excludedFilters?: string[];
+  initialConstraints?: InitialConstraints;
+}
+
+export const ModalProductFilterProvider: FC<ModalProductFilterProviderProps> = ({
+  children,
+  excludedFilters,
+  initialConstraints,
+}) => {
+  const { filterContext, filterVariables, filterChannel, clearFilters, hasActiveFilters } =
+    useModalProductFilter({
+      excludedFilters,
+      initialConstraints,
+    });
+
+  return (
+    <ConditionalFilterContext.Provider value={filterContext}>
+      <ModalProductFilterContext.Provider
+        value={{
+          filterContext,
+          filterVariables,
+          filterChannel,
+          clearFilters,
+          hasActiveFilters,
+        }}
+      >
+        {children}
+      </ModalProductFilterContext.Provider>
+    </ConditionalFilterContext.Provider>
+  );
+};
```

Comment:
todo: we should rewrite this to use jotai - but this would require bigger changes to our modals